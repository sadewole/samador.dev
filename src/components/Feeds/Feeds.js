import React from "react"
import { Link } from "gatsby"

const Feeds = ({ posts }) => (
  <div className="feed">
    {posts.map(post => (
      <article className="feed-item">
        <div className="feed-meta">
          <h2 className="feed-title">
            <Link className="feed-link" to={post.node.frontmatter.slug}>
              {post.node.frontmatter.title}
            </Link>
          </h2>
          <time className="feed-time" dateTime={post.node.frontmatter.date}>
            {post.node.frontmatter.date}
          </time>
        </div>
        <section>
          <p className="feed-description">{post.node.excerpt}</p>
        </section>
      </article>
    ))}
  </div>
)

export default Feeds
