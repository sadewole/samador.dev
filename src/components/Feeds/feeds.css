.feed-time {
  font-size: 14px;
  font-weight: 600;
  color: var(--color-accent);
  text-transform: uppercase;
}

.feed-title {
  font-size: 27px;
  margin-bottom: var(--spacing-2) !important;
}
