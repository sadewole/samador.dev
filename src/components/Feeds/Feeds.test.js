import React from "react"
import renderer from "react-test-renderer"

import Feeds from "./Feeds"

describe("Feeds", () => {
  const props = {
    posts: [
      {
        excerpt: "test excerpt",
        frontmatter: {
          slug: "test slug",
          title: "test title",
          date: "tesst date",
        },
      },
    ],
  }
  it("render correctly", () => {
    const tree = renderer.create(<Feeds {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
