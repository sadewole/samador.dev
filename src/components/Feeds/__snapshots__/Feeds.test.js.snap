// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feeds render correctly 1`] = `
<div
  className="feed"
>
  <article
    className="feed-item"
  >
    <div
      className="feed-meta mb-5"
    >
      <h2
        className="feed-title"
      >
        <a
          className="feed-link"
          href="test slug"
        >
          test title
        </a>
      </h2>
      <time
        className="feed-time"
        dateTime="tesst date"
      >
        tesst date
      </time>
    </div>
    <section>
      <p
        className="feed-description"
      >
        test excerpt
      </p>
    </section>
  </article>
</div>
`;
